src/internet/ipv4.rs

```diff
@@ -28,6 +29,7 @@ pub struct Ipv4Header {
     pub destination: [u8;4],
     /// Length of the options in the options_buffer in bytes.
     options_len: u8,
+    #[cfg_attr(feature = "serde", serde(skip), serde(default = "default_options_buffer"))]
     options_buffer: [u8;40]
```

Comment:
Same as with the `TcpHeader` & `Ipv6RawExtension` I would implement `Serialize` & `Deserialize` manually as options buffer is not required to be fully filled (see comments there).